Jint/Options.cs

```diff
@@ -431,6 +431,12 @@ public class ConstraintOptions
         /// </summary>
         public TimeSpan RegexTimeout { get; set; } = TimeSpan.FromSeconds(10);
 
+
+        /// <summary>
+        /// Maximum time allowed for unwrapping a Promise and getting its resolved/rejected value.
+        /// Defaults to 10 seconds.
+        /// </summary>
+        public TimeSpan UnwrapIfPromiseTimeout { get; set; } = TimeSpan.FromSeconds(10);
```

Comment:
Maybe name this just PromiseTimeout, UnwrapIfPromise is a techinical term and implementation detail.